incubator/hnc/pkg/forest/forest.go

```diff
@@ -214,6 +215,24 @@ func (ns *Namespace) clean() {
 	delete(ns.forest.namespaces, ns.name)
 }
 
+// UpdateAllowCascadingDelete updates if this namespace allows cascading deletion.
+func (ns *Namespace) UpdateAllowCascadingDelete(acd bool) {
+	ns.allowCascadingDelete = acd
+}
+
+// AllowsCascadingDelete returns if the namespace's or any of the owner ancestors'
+// allowCascadingDelete field is set to true.
+func (ns *Namespace) AllowsCascadingDelete() bool {
+	if ns.allowCascadingDelete == true {
+		return true
+	}
+	if !ns.IsOwned {
+		return false
+	}
+	// If the owner is missing, it will return the default false.
+	return ns.Parent().AllowsCascadingDelete()
+}
+
```

Comment:
@adrianludwin this is where the `ns.Ower` => `ns.IsOwned` change is.

Reply:
Thanks!